core/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -124,7 +124,13 @@ public Weight createWeight(Query query, boolean needsScores) throws IOException
                 profile.stopAndRecordTime();
                 profiler.pollLastElement();
             }
-            return new ProfileWeight(query, weight, profile);
+
+            if (weight instanceof ProfileWeight) {
```

Comment:
It's not possible to do this, because this would require `weight = super.createWeight(query, needsScores)` to run before `startTime()`, but this is the thing being timed.